react-client-app/src/components/UploadScanReport.jsx

```diff
@@ -0,0 +1,398 @@
+            // if(datasets_query.find(ds => ds.id == newDataset.id)){
+            //     setselectedDataset(newDataset)
+            // }
+            // else
+
+            if (!datasets_query.find(ds => ds.id == selectedDataset.id)) {
+                setselectedDataset({ name: "------" })
+            }
+            setLoadingDataset(false)
+            closeAddingInterface()
+
+            // Could add newly created dataset to dataset list manually like this rather than revalidating
+            // setDatasets(
+            //     ds => [{ name: "------" }, ...[newDataset, ...ds.filter(ds2 => ds2.id != undefined)].sort((a, b) => a.name.localeCompare(b.name))]
+            // )
+
+
+
+            setAlert({
+                hidden: false,
+                status: 'success',
+                title: 'Success',
+                description: 'New Dataset added'
+            })
+            onOpen()
+        }
+        catch (err) {
+            setAlert({
+                hidden: false,
+                status: 'error',
+                title: 'Could not add dataset',
+                description: err.statusText ? err.statusText : ""
+            })
+            closeAddingInterface()
+            onOpen()
+        }
+    }
+    const upload = async () => {
+        try {
+            if (selectedDataPartner.id == undefined) {
+                throw { statusText: "Please select a datapartner" }
+            }
+            if (selectedDataset.id == undefined) {
+                throw { statusText: "Please select a dataset" }
+            }
+            if (!scanReportName.current.value) {
+                throw { statusText: "Please choose a scan report name" }
+            }
+            if (!whiteRabbitScanReport) {
+                throw { statusText: "Please add a scan report file" }
+            }
+            if (whiteRabbitScanReport.name.split('.').pop() != "xlsx") {
+                throw { statusText: "You have attempted to upload a scan report which is not in XLSX format. Please upload a .xlsx file" }
+            }
+            if (dataDictionary && dataDictionary.name.split('.').pop() != "csv") {
+                throw { statusText: "You have attempted to upload a data dictionary which is not in csv format. Please upload a .csv file" }
+            }
+            // The tests on the backend will also trigger an error
+
+            let formData = new FormData()
+            await formData.append('parent_dataset', selectedDataset.id)
+            await formData.append('dataset', scanReportName.current.value)
+            await formData.append('scan_report_file', whiteRabbitScanReport)
+            await formData.append('data_dictionary_file', dataDictionary)
+            setUploadLoading(true)
+
+            const response = await postForm(window.location.href,formData)
+            // redirect if the upload was successful, otherwise show the error message
+            window.location.href = `${window.u}scanreports/`
```

Comment:
`window.u` will not work anymore. I think if you want to change just the path you use `window.location.pathname` right?
```suggestion
            window.location.pathname = `/scanreports/`
```